plutus-core/plutus-core/src/PlutusCore.hs

```diff
@@ -149,12 +135,13 @@ import PlutusCore.Rename
 import PlutusCore.Size
 import PlutusCore.TypeCheck as TypeCheck
 
-import UntypedPlutusCore.Evaluation.Machine.Cek.CekMachineCosts
-
 -- | Take one PLC program and apply it to another.
 applyProgram
     :: Monoid a
     => Program tyname name uni fun a
     -> Program tyname name uni fun a
     -> Program tyname name uni fun a
-applyProgram (Program a1 _ t1) (Program a2 _ t2) = Program (a1 <> a2) (defaultVersion mempty) (Apply mempty t1 t2)
+-- TODO: 'mappend' annotations, ignore versions and return the default one (whatever that means),
```

Comment:
Honestly, this function is super sketchy and should probably not exist.

Reply:
There's also

```haskell
applyProgram
    :: Monoid a
    => Program tyname name uni fun a
    -> Program tyname name uni fun a
    -> Program tyname name uni fun a
applyProgram (Program a1 t1) (Program a2 t2) = Program (a1 <> a2) (Apply mempty t1 t2)
```

in PIR and 

```haskell
applyProgram :: Program name uni fun () -> Program name uni fun () -> Program name uni fun ()
applyProgram (Program _ _ t1) (Program _ _ t2) = Program () (PLC.defaultVersion ()) (Apply () t1 t2)
```

in UPLC.

Amazingly enough, they all are different. We should sort this out.

Reply:
:vomiting_face:

Reply:
Almost certainly my fault :sweat_smile:

Reply:
Created PLT-224 for that.